dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -118,21 +118,30 @@ def _determine_pf_parts(fs, paths, gather_statistics, **kwargs):
     parts = []
     if len(paths) > 1:
         paths, base, fns = _sort_and_analyze_paths(paths, fs)
+
+        # Check if _metadata is in paths, and
+        # remove it if ignore_metadata_file=True
+        _metadata_exists = "_metadata" in fns
+        if _metadata_exists and ignore_metadata_file:
+            fns.remove("_metadata")
+            paths = [fs.sep.join([base, fn]) for fn in fns]
+            _metadata_exists = False
+
         if gather_statistics is not False:
             # This scans all the files, allowing index/divisions
             # and filtering
-            if "_metadata" not in fns:
-                paths_use = paths
+            if _metadata_exists:
+                paths_use = fs.sep.join([base, "_metadata"])
```

Comment:
I think I get confused here - don't we end up using _metadata even with ignore_metadata_file, if we asked for statistics? That's the inverse of my previous objection, but feels wrong this way around too, especially for gather_statistics=None

Reply:
>don't we end up using _metadata even with ignore_metadata_file, if we asked for statistics?

If `ignore_metadata_file=True`, we will not use the `_metadata` file.  However, we may and up gathering statistics manually.  We do not want `ignore_metadata_file` to be synonymous with `gather_statistics`.

To clarify further, I personally **don't** feel that the default behavior of `gather_statistics` should be `False` when the `_metadata` file is missing or ignored. If the user is not filtering, *and* is not setting and index, *and* is not aggregating files, then sure, default-False makes sense.

Reply:
Oh OK never mind - you set `_metadata_exists = False` a few lines above (so it no longer means what it says it does! :) ). That was my confusion - I agree with you on the behaviour we want.